library/misc/debug_renderers/src/main/java/org/quiltmc/qsl/debug_renderers/api/DebugFeature.java

```diff
@@ -0,0 +1,76 @@
+package org.quiltmc.qsl.debug_renderers.api;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.util.Identifier;
+import org.quiltmc.loader.api.minecraft.ClientOnly;
+import org.quiltmc.qsl.debug_renderers.impl.DebugFeaturesImpl;
+
+import java.util.Objects;
+
+public final class DebugFeature {
+	private final Identifier id;
+	private final boolean needsServer;
+
+	private DebugFeature(Identifier id, boolean needsServer) {
+		this.id = id;
+		this.needsServer = needsServer;
+	}
+
+	public boolean isEnabled() {
+		return DebugFeaturesImpl.isEnabled(this);
+	}
+
+	public boolean isEnabledOnServerAndClient(ServerPlayerEntity player) {
+		return DebugFeaturesImpl.isEnabled(this) && DebugFeaturesImpl.isEnabledForPlayer(player, this);
+	}
+
+	@ClientOnly
+	boolean isEnabledOnServerAndClient() {
```

Comment:
I don't like having an overload that is client-only. This would be really easy to use on accident

Reply:
True - come to think of it, the client overload isn't needed at all, and can just be folded into `shouldRender`.